### PR TITLE
update to support 3.1

### PR DESCRIPTION
**Pull request by Mossman1215**
modernising to the 3.1 api
adding ci support via github actions

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,15 +18,18 @@
...
-        "friendsofphp/php-cs-fixer": "^2.0"
+        "phpunit/phpunit": "^7",
+        "friendsofphp/php-cs-fixer": "^2.0",
+        "phpstan/phpstan": "^0.12.67"
     },
     "autoload": {
         "psr-0": {
-            "stojg": "src/"
+            "Silverstripe": "src/",
+            "Silverstripe/PingdomCLI": "src/PindomCLI"
```
Comment (indygriffiths) [inline composer.json]: Not sure whether you need both of these, I would remove line 32 and confirm you've spelt the directory correctly. Composer should then automatically pick up the namespace

Comment (indygriffiths) [reply composer.json]: PSR-4 states your directory name must match the namespace. In your code you're using `Silverstripe\Pingdom`, so the directory should be `src/Pingdom`

File: src/Silverstripe/PingdomCLI/GetChecks.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+use Silverstripe\Pingdom\Api;
+use Symfony\Component\Console\Helper\Table;
+
+class GetChecks extends Command
+{
+    // the name of the command (the part after "bin/console")
+    protected static $defaultName = 'get-checks';
+    private $pingdomToken;
+    protected function configure()
+    {
```
Comment (indygriffiths) [inline src/Silverstripe/PingdomCLI/GetChecks.php]: Have you run `vendor/bin/php-cs-fixer`? Syntax is all over the place in this file

File: src/Silverstripe/PingdomCLI/GetChecks.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }else{
+        $output->writeln('No authentication token provided');
+        $output->writeln('exiting');
+        return 1;
+    }
+      $pingdom = new Api($this->pingdomToken);
+      $checks = $pingdom->getChecks();
+      if($input->getOption('json')){
+        $output->writeln(\json_encode($checks));
+        return 0;
```
Comment (indygriffiths) [inline src/Silverstripe/PingdomCLI/GetChecks.php]: If this is a newer version of symfony/console, use `Command::SUCCESS` constant, likewise `Console::FAILURE` for the failure return codes.

File: .github/workflows/php.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    - name: Install dependencies
+      if: steps.composer-cache.outputs.cache-hit != 'true'
+      run: composer install --prefer-dist --no-progress --no-suggest
+
+    # Add a test script to composer.json, for instance: "test": "vendor/bin/phpunit"
+    # Docs: https://getcomposer.org/doc/articles/scripts.md
+
+    - name: Run test suite
+      run: vendor/bin/phpunit
```
Comment (indygriffiths) [inline .github/workflows/php.yml:38]: Also make sure php-cs-fixer is being run during these tests. See the circleci config for Dash for an example to exit 1 if the syntax is wrong

File: bin/pingdom-cli [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env php
+<?php
+
+require __DIR__."/../vendor/autoload.php";
+
+use Symfony\Component\Console\Application;
+use Silverstripe\PingdomCLI\GetChecks;
+$application = new Application();
```
Comment (indygriffiths) [inline bin/pingdom-cli]: Set an application name in the argument, e.g `new Application('Pingdom CLI');`

File: src/Silverstripe/PingdomCLI/GetChecks.php [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+class GetChecks extends Command
+{
+    // the name of the command (the part after "bin/console")
+    protected static $defaultName = 'get-checks';
+    private $pingdomToken;
+    protected function configure()
+    {
+      $this->addOption('json')
+      ->addArgument('api-key',InputArgument::OPTIONAL,'your oauth token','');
```
Comment (indygriffiths) [inline src/Silverstripe/PingdomCLI/GetChecks.php]: Add a `->setDescription()` chained function to outline what the command does. I'm a big believer that any new CLI tools we make should be able to document themselves

File: src/Silverstripe/PingdomCLI/GetChecks.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputArgument;
+use Silverstripe\Pingdom\Api;
+use Symfony\Component\Console\Helper\Table;
+
+class GetChecks extends Command
+{
+    // the name of the command (the part after "bin/console")
+    protected static $defaultName = 'get-checks';
```
Comment (indygriffiths) [inline src/Silverstripe/PingdomCLI/GetChecks.php]: Our `symfony/console` convention would mean `checks:get` is a better approach. That way we can add future `checks:*` commands under one command namespace

File: composer.json (hunk 1 above)
Comment (indygriffiths) [reply composer.json]: Actually now that I think about it, I think you need:

```json
"autoload": {
        "psr-0": {
                "Silverstripe": "src/"
        }
}
```
and structure you code in:
```
src/Silverstripe/Pingdom/Api/Client.php
src/Silverstripe/Pingdom/Api/Exception/ClientErrorException.php etc etc
src/Silverstripe/Pingdom/Command/Checks/Get.php
```
Note that I've renamed Pingdom.php to Client.php, and GetClient.php to Get.php

Comment (Mossman1215) [reply composer.json]: doesn't the following
```
"autoload": {
        "psr-0": {
                "Silverstripe": "src/"
        }
}
```
 alias "src/" as "silverstripe"?

File: composer.json [hunk 8]
```diff
@@ -1,8 +1,12 @@
 {
-    "name": "stojg/pingdom-api",
+    "name": "silverstripe/pingdom-api",
```
Comment (indygriffiths) [inline composer.json]:
```suggestion
    "name": "silverstripeltd/pingdom-api",
```

File: src/Silverstripe/Pingdom/Command/Checks/Get.php [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    protected function configure()
+    {
+        $this->setDescription('get all pingdom checks attached to the provided account token')
+        ->addOption('json')
+        ->addArgument('api-key', InputArgument::OPTIONAL, 'your oauth token', '');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (strlen($input->getArgument('api-key')) > 0) {
```
Comment (indygriffiths) [inline src/Silverstripe/Pingdom/Command/Checks/Get.php]:
```suggestion
        if (count($input->getArgument('api-key')) > 0) {
```

Comment (Mossman1215) [reply src/Silverstripe/Pingdom/Command/Checks/Get.php]: dang first google result proves wrong
thought it was count() :smile:

**Review by indygriffiths — APPROVED** [review]

**Pull request merged**